### ISSUE TITLE
Multi-label classification

### ISSUE THREAD

**Issue by dettmar** [manujosephv/pytorch_tabular#70]
**Is your feature request related to a problem? Please describe.**
I see in the examples that multi-label classification isn't implemented yet. Is that still the case? And if so, is there a timeline for when this will be implemented?

**Describe the solution you'd like**
Many tabular problems require more than one-class classification. Since it's doable in vanilla PyTorch, I think it would be a fantastic addition to Tabular as well, since it makes training a lot more convenient and streamlined.

**Describe alternatives you've considered**
Wrapping a classifier in something like sklearn's [MultiOutputClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html) introduces many issues such as lack of native validation set, early stopping, etc.

**Additional context**

**Comment by numomcmc** [comment]
Hi

Any chance support for multi-label classification will be provided in the near future?  Thanks!!

**Comment by richarddushime** [comment]
Hi @fkiraly 
can i take on this ?
Thanks

**Comment by Jay2006sawant** [comment]
Hey @fkiraly! I would like to work on this issue

**Comment by Jayant-kernel** [comment]
Hi @dettmar and maintainers @fkiraly 

I'd like to work on adding multi-label classification support to pytorch_tabular!

My plan:
1. Add a `MultiLabelClassificationTask` task type in the task config
2. Handle multi-hot target encoding in the DataModule
3. Use `BCEWithLogitsLoss` as the default loss for multi-label tasks
4. Add appropriate metrics (e.g., hamming loss, F1 per label)
5. Include tests and a small example notebook

if this sounds can i proceed with PR
